src/components/Card/index.jsx

```diff
@@ -1,21 +1,19 @@
 import React, { useState } from 'react';
 import './styles.css';
-import { genresList } from '../../services';
+import {
+  fetchFrom,
+  setInStorage,
+  getFromStorage,
+  filterByReference,
+} from '../../services';
 import StarRating from '../StarRating';
 
-function Card({ movie, isPoster }) {
+function Card({ movie, isPoster, setMovieInfo }) {
   const movieImagePath = isPoster ? movie.poster_path : movie.backdrop_path;
   const [infoVisible, setInfoVisible] = useState(false);
+  const genresLabelList = filterByReference(getFromStorage('genresList'), movie.genre_ids).slice(0, 3);
   let timeOutId = null;
 
-  const filterByReference = (arr1, arr2) => {
-    let res = [];
-    res = arr1.filter((el) => arr2.find((element) => element === el.id));
-    return res.slice(0, 3);
-  };
-
-  const genresLabelList = filterByReference(genresList, movie.genre_ids);
-
   function showInfo() {
     timeOutId = setTimeout(() => {
```

Comment:
instead of creating a delay with a setTimeout will be better to make an animation with css